ds13/code/modules/mob/living/psychosis.dm

```diff
@@ -0,0 +1,182 @@
+/**
+*
+* Psychosis
+* @author Kmc2000
+* @version 11/11/2019
+* A series of procs which model insanity. You will heal insanity damage slowly over time, but if youre exposed to traumatic events then expect to get a bit unhinged. Finally, the effects are capped on a cooldown, so that necros don't permanently debilitate you. This is meant to be a slippery slope, not a cliff you jump off of.
+*
+* Important methods:
+* adjustPsychosisLoss(amount) - Use this to deal psychosis damage to a target.
+*
+*
+*/
+
+
+// Psychosis side effects:
+// ========
+
+
+/datum/medical_effect/irrational_laughter
+	name = "Irrational laughter"
+	triggers = list() //Triggered only by psychosis
+	cures = list(/datum/reagent/synaptizine) //Cured with brain meds
+	cure_message = "Life doesn't seem quite as absurd as it used to."
+
+/datum/medical_effect/irrational_laughter/on_life(mob/living/carbon/human/H, strength)
+	var/obj/item/organ/external/head/head = H.get_organ("head")
+	if(istype(head))
+		switch(strength)
+			if(1 to 10)
+				H.custom_pain("You suddenly feel an urge to laugh.",0, affecting = LOWER_TORSO)
+			if(11 to 30)
+				H.custom_pain("You burst out laughing!",1, affecting = LOWER_TORSO)
+			if(31 to INFINITY)
+				H.custom_pain("You burst out into an uncontrollable fit of laughter!",1, affecting = LOWER_TORSO)
+
+/datum/medical_effect/irrational_laughter/manifest(mob/living/carbon/human/H) //Override to move away from a reagent based trigger to an event based one.
+	for(var/R in cures) //If they have a cure inside of them, fail to manifest.
+		if(!H.reagents || H.reagents.has_reagent(R))
+			return FALSE
+	return TRUE
+
+// =======
+
+/**
+*
+*
+* A proc which handles fluff FX for people going mad.
+* @Param stage : How severe the psychosis is. This is a scaling value based on the mob's current psychosis damage (which is increased via witnessing traumatic events, or being touched by necros).
+*
+*/
+
+/mob/living/carbon/human/proc/set_psychosis(var/stage, var/override)
```

Comment:
I'm not clear on the purpose of this function at all, it doesn't seem to use the granular psychosis loss which is healed and set elsewhere, and the effects should probably not be embedded right here